File: native/macos/SkillFlow/Sources/Backup/BackupModels.swift
import Foundation

struct RemoteFile: Codable, Identifiable, Equatable {
    let path: String
    let size: Int64
    let isDir: Bool
    let action: String?

    var id: String { path }
}

struct BackupResolveConflictParams: Encodable {
    let useLocal: Bool
}

File: native/windows/SkillFlow/SkillFlow/Backup/BackupModels.cs
using System.Text.Json.Serialization;

namespace SkillFlow.Backup;

public sealed class RemoteFile
{
    [JsonPropertyName("path")]
    public string Path { get; set; } = string.Empty;

    [JsonPropertyName("size")]
    public long Size { get; set; }

    [JsonPropertyName("isDir")]
    public bool IsDir { get; set; }

    [JsonPropertyName("action")]
    public string? Action { get; set; }
}

public sealed class BackupResolveConflictParams
{
    [JsonPropertyName("useLocal")]
    public bool UseLocal { get; set; }
}
